ql/src/java/org/apache/hadoop/hive/ql/optimizer/lineage/Generator.java

```diff
@@ -59,11 +68,63 @@ public class Generator extends Transform {
 
   private static final Logger LOG = LoggerFactory.getLogger(Generator.class);
 
-  private final Set<String> hooks;
-  private static final String ATLAS_HOOK_CLASSNAME = "org.apache.atlas.hive.hook.HiveHook";
+  public static Generator fromConf(HiveConf conf) {
+    return new Generator(createFilterPredicateFromConf(conf));
+  }
+
+  private static final String ALL = "ALL";
```

Comment:
it relates to my previous comment at HiveConf about the possible values. At HiveConf, it is All, not ALL.